src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py

```diff
@@ -1201,7 +1062,23 @@ def generate_power_spec_commands(self) -> List[str]:
             self.logger.error(
                 "Invalid power specification type '{tpe}'; only 'cpf' or 'upf' supported".format(tpe=power_spec_type))
             return []
-        power_spec_contents = str(self.get_setting("vlsi.inputs.power_spec_contents"))  # type: str
+
+        power_spec_contents = ""  # type: str
+        power_spec_mode = str(self.get_setting("vlsi.inputs.power_spec_mode"))  # type: str
+        if power_spec_mode == "empty":
+            return []
+        elif power_spec_mode == "auto":
```

Comment:
Did we settle on the difference between `auto` and `generate`?

Reply:
I went with auto because more `*_mode` variables use it. I do think this is closer to `auto` than `generate` assuming `auto` means Hammer does everything and `generate` means you need to setup Hammer to be able to generate.
The default supplies are sane, so its really about the `type` choice being populated automatically that would make it full-auto.